### PR TITLE
fixing issues with continuation channels not working with writes, and improving .env dx in cli

### PR DESCRIPTION
**Pull request by xav-db**
<h3>Greptile Summary</h3>

This PR fixes a critical hang issue with write operations that require async IO by changing continuation channels from pool-level shared channels to per-request channels, and improves CLI developer experience with explicit .env file loading and user feedback.

## Key Changes

**Core Fix - Writer Continuation Channels:**
- Changed from shared pool-level continuation channel to per-request continuation channels in `start_writer()`
- Writer thread now creates `(cont_tx, cont_rx)` for each request, passes `cont_tx` to handler, then polls `cont_rx` until disconnected
- Prevents hangs when multiple write requests with async IO arrive concurrently by ensuring each request's continuations complete before processing next request
- Previous implementation used `try_recv()` which could miss pending continuations when processing next request

**CLI .env Improvements:**
- Explicit two-stage .env loading: project root first (base config), then queries directory (overrides)
- Added informative `print_info` messages showing which .env files are loaded
- Improves developer experience by making environment configuration transparent

**Testing Infrastructure:**
- Added 950+ lines of comprehensive continuation channel tests covering writer priority, parallel writes, read/write mixing, and edge cases
- Fixed worker pool concurrency tests by properly registering route handlers
- Serialized all LMDB stress tests with `#[serial(lmdb_stress)]` to prevent interference
- Updated CI workflows to run all tests including concurrency tests (now safe with serialization)

**Dependency Updates:**
- Upgraded `flume` from 0.11.1 to 0.12.0 across all crates
- Removed `eventual-fairness` feature (no longer needed with per-request channels)
- Added `serial_test` crate for test serialization

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| helix-db/src/helix_gateway/worker_pool/mod.rs | Fixed continuation channel handling for writes by creating per-request channels instead of shared pool-level channel, preventing hangs with concurrent async operations |
| helix-cli/src/docker.rs | Enhanced .env loading with explicit priority: project root loaded first, then queries directory overrides, with informative user feedback for each |
| helix-db/src/helix_gateway/tests/worker_pool_tests.rs | Added comprehensive continuation channel tests including writer priority, parallel writes with continuations, and mixed read/write scenarios |
| .github/workflows/db_tests.yml | Simplified to run all tests including concurrency tests now that LMDB interference is prevented via serial_test |
| helix-db/Cargo.toml | Updated flume to 0.12.0, removed eventual-fairness feature, added serial_test dependency |

</details>

</details>

<details><summary><h3>Sequence Diagram</h3></summary>

```mermaid
sequenceDiagram
    participant Client
    participant WorkerPool
    participant WriterThread
    participant AsyncRuntime
    participant Handler
    
    Note over WorkerPool,WriterThread: Writer Thread Loop
    
    Client->>WorkerPool: Write Request
    WorkerPool->>WriterThread: Forward to write_tx channel
    
    Note over WriterThread: Create per-request<br/>continuation channel
    WriterThread->>WriterThread: let (cont_tx, cont_rx) = bounded(1)
    
    WriterThread->>Handler: request_mapper(req, cont_tx)
    Handler->>Handler: Process request
    
    alt Handler needs async IO
        Handler->>AsyncRuntime: Spawn async task (clone cont_tx)
        Handler-->>WriterThread: Return IoNeeded error
        
        Note over WriterThread: Drop original cont_tx
        WriterThread->>WriterThread: drop(cont_tx)
        
        Note over WriterThread: Poll continuation channel
        WriterThread->>WriterThread: while let Ok((ret_chan, cfn)) = cont_rx.recv()
        
        AsyncRuntime->>AsyncRuntime: Perform async IO
        AsyncRuntime->>WriterThread: Send continuation via cont_tx.send_async()
        
        WriterThread->>WriterThread: Execute continuation: cfn()
        WriterThread->>Client: Send response via ret_chan
        
        Note over WriterThread: Channel disconnects when<br/>async future drops cont_tx
        WriterThread->>WriterThread: cont_rx.recv() returns Err
    else Handler completes immediately
        Handler-->>WriterThread: Return response
        WriterThread->>Client: Send response
    end
    
    Note over WriterThread: Move to next request<br/>in queue
    WriterThread->>WorkerPool: Ready for next write request
```
</details>

### CODE REVIEW

**Comment by xav-db** [comment]
@greptile

File: helix-cli/src/docker.rs [hunk 1]
```diff
@@ -87,8 +87,14 @@ impl<'a> DockerManager<'a> {
...
-        // Load .env file (silently ignore if it doesn't exist)
-        let _ = dotenvy::dotenv();
+        // Load .env from project root first (base configuration)
+        let root_env = self.project.root.join(".env");
+        let _ = dotenvy::from_path(&root_env);
+
+        // Load .env from db/queries directory (overrides project root)
+        let queries_dir = self.project.root.join(&self.project.config.project.queries);
+        let db_env = queries_dir.join(".env");
+        let _ = dotenvy::from_path_override(&db_env);
```
Comment (matthewsanetra) [inline helix-cli/src/docker.rs]: This should have info logging to make it clearer to the user.

e.g. (wording can be changed to be clearer)
```
[INFO] Using .env in project root
[INFO] Overriding environment variables from {path}/.env
```

Easier debugging for users if they accidentally leave MY_ENV=<empty> in the db/queries dir and it unsets it

File: helix-db/src/helix_gateway/worker_pool/mod.rs [hunk 2]
```diff
@@ -236,39 +236,51 @@ impl Worker {
             // Set thread local context, so we can access the io runtime
             let _io_guard = io_rt.enter();
 
-            // Simple loop for the writer - no parity needed since it's a single thread
+            // Use Selector to wait on both channels simultaneously.
+            // This is required because with a single writer thread, we can't use the
+            // parity trick that read workers use (which requires multiple threads).
```
Comment (matthewsanetra) [inline helix-db/src/helix_gateway/worker_pool/mod.rs]: Correct, but continuation functions *must* be given priority otherwise it'll hang when another write request is sent before another write requiring continuation has finished. The selector must either not have fairness or explicitly poll continuation channel before this selector.

Test suite should be updated to catch the error of what I described above, since this will hang

Comment (matthewsanetra) [reply helix-db/src/helix_gateway/worker_pool/mod.rs]: On second thought it should not use a selector ever. Writes are single threaded so: only read the request channel, make a continuation channel, keep polling it until its dropped, and only then continue reading from request channel

**Comment by xav-db** [comment]
@greptile
